### ISSUE TITLE
favorite stops working

### ISSUE THREAD

**Issue by dttpy** [owncloud/core#13287]
### Steps to reproduce
UNKNOWN
It happened just accidentally. The steps to reproduce hasn't be found yet. 
I have just tried to create folder, tag them as favorite, rename, delete the folder from "All files" or "favorite" to trash bin a lot of times, and try to delete them from trash bin

### Expected behaviour
favorite works 

### Actual behaviour
favorite stops working

And in the DB, it looks like that the favorite tags is still saved in the trash bin.

```
sqlite> select path from oc_vcategory_to_object,oc_filecache where categoryid=16 and objid=fileid;
files/infinite scrolling
files_trashbin/files/111.d1417131063
files_trashbin/files/234.d1417131533

```

### Server configuration
**Operating system:** Ubuntu 13.10

**Web server:** apache 2.4.6 (Ubuntu)

**Database:** SQLite

**PHP version:** 5.5.3-1ubuntu2.3

**ownCloud version:** owncloud 8 git master

**Updated from an older ownCloud or fresh install:** 

**List of activated apps:** standard apps of owncloud

**The content of config/config.php:**

**Are you using external storage, if yes which one:** no
**Are you using encryption:** yes

### Client configuration
**Browser:** firefox 34.0.5

**Operating system:** ubuntu 13.10

#### ownCloud log (data/owncloud.log)
```
{"reqId":"d05a882d24c7b1b22c0e62fa8d092b09","remoteAddr":"192.168.1.34","app":"no app in context","message":"Accessing the server container is deprecated. Use type annotations to inject core services instead!","level":0,"time":"2014-11-27T23:37:24+00:00","method":"GET","url":"\/owncloud\/index.php\/apps\/files\/api\/v1\/tags\/_$!%3CFavorite%3E!$_\/files"}
{"reqId":"d05a882d24c7b1b22c0e62fa8d092b09","remoteAddr":"192.168.1.34","app":"core","message":"OC\\Tags::__construct, tags: Array\n(\n    [0] => OC\\Tagging\\Tag Object\n        (\n            [owner:protected] => admin\n            [type:protected] => files\n            [name:protected] => _$!<Favorite>!$_\n            [id] => 16\n            [_updatedFields:OCP\\AppFramework\\Db\\Entity:private] => Array\n                (\n                )\n\n            [_fieldTypes:OCP\\AppFramework\\Db\\Entity:private] => Array\n                (\n                    [id] => integer\n                )\n\n        )\n\n)\n","level":0,"time":"2014-11-27T23:37:24+00:00","method":"GET","url":"\/owncloud\/index.php\/apps\/files\/api\/v1\/tags\/_$!%3CFavorite%3E!$_\/files"}
{"reqId":"d05a882d24c7b1b22c0e62fa8d092b09","remoteAddr":"192.168.1.34","app":"core","message":"OC\\Tags::__construct, tags: Array\n(\n    [0] => OC\\Tagging\\Tag Object\n        (\n            [owner:protected] => admin\n            [type:protected] => files\n            [name:protected] => _$!<Favorite>!$_\n            [id] => 16\n            [_updatedFields:OCP\\AppFramework\\Db\\Entity:private] => Array\n                (\n                )\n\n            [_fieldTypes:OCP\\AppFramework\\Db\\Entity:private] => Array\n                (\n                    [id] => integer\n                )\n\n        )\n\n)\n","level":0,"time":"2014-11-27T23:37:24+00:00","method":"GET","url":"\/owncloud\/index.php\/apps\/files\/api\/v1\/tags\/_$!%3CFavorite%3E!$_\/files"}
{"reqId":"d05a882d24c7b1b22c0e62fa8d092b09","remoteAddr":"192.168.1.34","app":"PHP","message":"Argument 1 passed to OCA\\Files\\Helper::formatFileInfo() must implement interface OCP\\Files\\FileInfo, boolean given, called in \/var\/www\/owncloud\/apps\/files\/controller\/apicontroller.php on line 97 and defined at \/var\/www\/owncloud\/apps\/files\/lib\/helper.php#107","level":0,"time":"2014-11-27T23:37:24+00:00","method":"GET","url":"\/owncloud\/index.php\/apps\/files\/api\/v1\/tags\/_$!%3CFavorite%3E!$_\/files"}
{"reqId":"d05a882d24c7b1b22c0e62fa8d092b09","remoteAddr":"192.168.1.34","app":"PHP","message":"Call to a member function getMimetype() on a non-object at \/var\/www\/owncloud\/apps\/files\/lib\/helper.php#49","level":3,"time":"2014-11-27T23:37:24+00:00","method":"GET","url":"\/owncloud\/index.php\/apps\/files\/api\/v1\/tags\/_$!%3CFavorite%3E!$_\/files"}
```

#### Browser log

**Comment by dttpy** [comment]
@PVince81

**Comment by PVince81** [comment]
Also could not reproduce the issue when doing many deletion manipulations: delete as owner, as recipient, etc.

It is very strange how you managed to get "files_trashbin" files tagged. Normally the file ids change when deleting files, so it is not possible for tagged file to stay tagged when moved into the trashbin.

Or maybe there is some particular file operation that does strive and keep the file id.

About the exception: the boolean value is likely to be "false". Normally the code throws exceptions on error. But the Tags API returns "false" when there is an error... so it is likely that the tags API was called with an invalid value which causes it to return "false" and trigger the error that we see in the log above. The invalid value could be caused by the trashbin part.

Needs further research.

**Comment by PVince81** [comment]
Ok, I manually changed the database to purposefully point to a trashbin item.
And now I can see the same error.

So even though we don't know yet what triggered it, I can at least make the code more robust and still make the favorites list work, even with that strange error.

**Comment by PVince81** [comment]
Fix for the empty favorites list is here: https://github.com/owncloud/core/pull/13291
Only testable if the database already contains a tag entry pointing to files_trashbin.

**Comment by DeepDiver1975** [comment]
@PVince81 #13291 was merged - anything more we can do? You stated #13291 is a partial fix for this issue. THX

**Comment by PVince81** [comment]
As long as we don't know the steps how to move a file to trashbin without changing its file id (mysterious case), there isn't much we can do.

Let's close for now.

If @dttpy manages to reproduce the situation again we can reopen.
